### DISCUSSION TITLE
Casadi could not find ipopt plugin [Docker Container]

### DISCUSSION THREAD

**Discussion by nthanapaisal** [casadi/casadi#3103] [Q&A] [answered]
Hello,

I am currently developing an application that uses casadi and uses ipopt solver. However I am encountering an issue when trying to run my application on docker container and it could not find ipopt plugin. I however wrote unittests and tested locally on my mac before then and all I had to do was pip install casadi and it worked. 

Error Message: 

```
    Fp = nlpsol("Fp", "ipopt", nlp, opts)
  File "/opt/conda/lib/python3.9/site-packages/casadi/casadi.py", line 28924, in nlpsol
    return _casadi.nlpsol(*args)
RuntimeError: .../casadi/core/plugin_interface.hpp:292: Plugin 'ipopt' is not found.
```

- This is the base image for my docker container: `continuumio/anaconda3:2022.05`
- I have tried 3.5.5 casadi but `ERROR: No matching distribution found for casadi==3.5.5`
__________________________________________________________________________________________
I wonder ...

1. Why did I work on my local but not in my docker container? 
2. What is the correct way to install ipopt on the docker container? 

Thanks in advance

**Comment by jgillis** [comment]
Probably caused by a bug in 3.6.0 binaries where libgfortran failed to ship ( https://github.com/casadi/casadi/issues/3102 ). Expect a fix in 3.6.1 to be released next week.

**Comment by jgillis** [comment]
Do you have a link to the Dockerfile? It coukd still be other issues.

**Comment by jgillis** [comment, marked as answer]
Seems to be the missing gfortran dependency alright (to be fixed in 3.6.1).\
In the meantime, I suggest: `sudo apt install libgfortran5`, and use casadi `3.6.0`
